frappe/email/doctype/contact/contact.py

```diff
@@ -135,3 +135,13 @@ def contact_query(doctype, txt, searchfield, start, page_len, filters):
 			'link_doctype': filters.get('link_doctype'),
 			'link_name': filters.get('link_name')
 		})
+
+def contact_links(doctype, txt, searchfield, start, page_len, filters):
+	if not txt: txt = ""
+	txt = txt.lower()
+	return [[d] for d in get_link_doctypes() if txt in d.lower()]
```

Comment:
@RobertSchouten ,

include the txt (search) txt in the SQL query so we will get the filtered result instead of filtering result after. Please find the examples in queries.py